protocol/app/upgrades/v5.2.0/upgrade.go

```diff
@@ -0,0 +1,61 @@
+package v_5_2_0
+
+import (
+	"context"
+	"fmt"
+
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/dydxprotocol/v4-chain/protocol/lib"
+	clobtypes "github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+	vaultkeeper "github.com/dydxprotocol/v4-chain/protocol/x/vault/keeper"
+)
+
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	clobKeeper clobtypes.ClobKeeper,
+	vaultKeeper vaultkeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx context.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		sdkCtx := lib.UnwrapSDKContext(ctx, "app/upgrades")
+		sdkCtx.Logger().Info(fmt.Sprintf("Running %s Upgrade...", UpgradeName))
+
+		// Migrate x/clob.
+		clobKeeper.UnsafeMigrateOrderExpirationState(sdkCtx)
```

Comment:
Is there any preupgrade state needed to ensure the logic in this function is executed?

Reply:
vaults quoting should be enough. however, let me add explicit logic to place long term orders